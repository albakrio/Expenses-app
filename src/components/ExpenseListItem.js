import React from 'react';
import {connect} from 'react-redux'
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({amount,createdAt, description, dispatch, id}) => {
	
	return (
		<div>
			<h1>{description}</h1>
			<p>
				{amount} - {createdAt}
			</p>
			<button onClick={()=>{dispatch(removeExpense({id}))}}>Delete</button>
		</div>
	);
};

export default connect()(ExpenseListItem);
