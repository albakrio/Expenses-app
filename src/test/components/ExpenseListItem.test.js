import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render ExpenseListItem component', () => {
	// const mock = {
	// 	id: 1,
	// 	description: 'rent',
	// 	amount: 10095,
	// 	createdAt: 0,
	// 	note: ''
	// };

	const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
	expect(wrapper).toMatchSnapshot();
});
