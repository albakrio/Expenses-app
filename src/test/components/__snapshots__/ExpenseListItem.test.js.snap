// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem component 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h1>
      rent
    </h1>
  </Link>
  <p>
    $10.10
    -
    December 31st, 1969
  </p>
</div>
`;
